### ISSUE TITLE
Loading of css3.css

### ISSUE THREAD

**Issue by JN-Jones** [mybb/mybb#2108]
Whenever css3.css stylesheet is empty, it is still loaded by the browser. Since we can't remove it from the theme's stylesheet-list(only revert it), wouldn't it be better to check if the file wasn't empty, then add it to the list of stylesheets to be loaded. 

There would be less trips to the server that way. 

(Why did you(the team) even decide to keep all css3 code in a separate file)

Original thread: [Loading of css3.css](http://community.mybb.com/thread-169131.html)

**Comment by Sama34** [comment]
The only possible way to currently disable an style-sheet from loading is to assign it to an invalid page.

**Comment by Destroy666x** [comment]
Yes, hence it was pushed. Some kind of on/off switch or something similar would be nice.

**Comment by JN-Jones** [comment]
@mybb/developers What's the preferred solution to this? A manual on/off switch which may also be helpful for testing purposes or simply avoid loading empty stylesheets?

**Comment by euantorano** [comment]
A manual on/off switch would be my preferred approach for definite.

**Comment by wagey** [comment]
Shouldn't both be considered?

1. Not allowing empty stylesheets to be loaded.
2. The option of enabling or disabling a stylesheet that isn't empty.

**Comment by JN-Jones** [comment]
Yeah, both would be possible and likely some sort of combination would be nice (eg setting the switch to "no" as soon as an empty stylesheet is saved). Haven't had time to work on this yet.

**Comment by Sama34** [comment]
I think it could be better if we add the switch option and disable empty style sheets upon saving instead of while loading.

**Comment by effone** [comment]
@euantorano @Sama34 This is being odd as I can't see generated form controls has options to set disabled. I need to either add that functionality or allow the user to set an empty css active and on submission validate and if the css is empty just throw an error and reset the value to inactive (no) again.

Suggestions?

**Comment by Sama34** [comment]
The addition of _file_get_contents_ is what would concern me.

I would probably add the active switch to the options drop down instead, and add a bullet image to visually identify its state.

And probably don't check for its emptyness (I contradict myself, lol). Rather, let the upgrade system deactivate them once and then it would be up to the user. Forcing disabling could bring issues, who knows.

**Comment by effone** [comment]
@JN-Jones @Sama34 @euantorano 
Alright, the PR is ready.

I have skipped the part of checking css file while saving and disabling as the process is completely interacting with a different table where active / order doesn't belong. If I do that here I have to repeat a whole lot of code, you may say the total process of actorder post action. In that scenario transferring it to a function is better idea, but current behaviour appeared okay to me, so I leave it here saving bunch of process repetition and database interaction, for suggestions:

Current scenario:
User can create blank css and attach to any page. Default state of new / unknown css is active.
If user activate a blank css it will accept. No validation. But during loading if it is found empty it will not load.
As soon as user put contents in blank css it will start loading.
Disabled CSS ofcourse will not noad anyway.

**Comment by Sama34** [comment]
I think you are over thinking it. Users don't need to deactivate multiple style sheets at the same time. Just add a new option within the _Options_ dropdown and a new post call. No need to check for anything but within the upgrade file at most.
